### ISSUE TITLE
Rate Limit Reached for emojis above 50+

### ISSUE THREAD

**Issue by FaizaanMeghani** [Jisagi/Discord-guild-copy#48]
Rate limit is reached at the 50th emoji added and then console is stucked.
Nothing happening. Waited around 10min. :c

**Comment by Jisagi** [comment]
The problem is, that discord does not list a specific ratelimit for emojis. To quote their api doc: *"Routes for controlling emojis do not follow the normal rate limit conventions."*.
Stoping exactly after 50 does sound like a fixed limit X per time Y (like 50 per hour), so there isn't anything I can do about that. As soon as the limit is reached, you'll encouter ratelimiting. Other endpoints are limit by amount/s or amount/min, which would've been preferable here.

Your waiting time of ~10min indicates an at least 10-15min large windows where the limit 50 is set. In theory you could wait even longer and see when it start to work again and thereby determine the actual timeframe in which the limit is applied. That won't solve the underlying problem though.

The only thing I can advise here is, restore the remaining emojis manually. You shouldn't be forced to restore your emojis *taht* often anyway, so doing it manually would be a one time thing. 

 I did already encounter problem back when only 50 emojis where the limit, when animated ones didn't exist yet. Now that the nitro boosts increase these limits even more, the problem will only increase. I will probably add a notice into the readme for this scenario, since I can't really fix not cirumvent this limit. The latter one would be highly risky anyway and could lead to further problem down the line.

TL;DR: Sadly not fixable, restore emojis manually

**Comment by FaizaanMeghani** [comment]
Yes I understand :c 

For now, is it possible to add a "skip" function when the emojis limit is reached so the bot can continue his actions and finish ? 
I just want to add the less emojis possible manually if possible.

Thank's

**Comment by Jisagi** [comment]
I don't see any possibility to reliably check if the ratelimit is reached at a certain point in time. Since the emoji creation is the second last step before the ban creation, you could just force the script to stop. If you don't need any bans restored, this would be the easiest option. Everything the script runs after the bans is not actually necessary and can be skipped completely without harm.

**Comment by FaizaanMeghani** [comment]
Okay :0
Thank's
